### PR TITLE
[TP] Build against Eclipse 4.25-I-builds

### PR DESCRIPTION
**Pull request by HannesWell**
If we release the next version of M2E before the Eclipse 2022-09 release it will be definitely targeted to that version. Therefore I think it is OK to build against 4.25-I-builds already now.

@mickaelistria (if you are reading during your vacation), @laeubi what do you think?

@mihnita can the MavenLaunchDelegegate be simplified further? IIRC you said that there is already a check per console if coloring is supported/enabled? Or is that only planned?

### CODE REVIEW

File: .gitmodules
Comment (HannesWell) [inline .gitmodules:3]: @mihnita this way we can test changes in the m2e-core-tests submodule before merging them.
It is a bit more complicated and requires at least two verification-builds (if one wants to always have green builds), but it works.
Of course this has to be revered before this PR is submitted.

File: target-platform/target-platform.target
Comment (HannesWell) [inline target-platform/target-platform.target:16]: This is necessary since https://github.com/eclipse-pde/eclipse.pde/pull/269 is only available in the I-Builds and not even in a Milestone, but `org.eclipse.pde.ds.lib` already has been removed from the I-build repo.

**Comment by mihnita** [comment]
> @mihnita can the MavenLaunchDelegegate be simplified further?
I can't imagine how. If we require a newer version of org.eclipse.ui.console, one that has ANSI, we can remove `isAnsiSupportInstalled()`, but nothing other than that.

> IIRC you said that there is already a check per console if coloring is supported/enabled? 

There is none.
Right now the enable / disable status is global (for same for all consoles).
And that is the one accessed with `preferencesService.getBoolean(...)` in `isAnsiProcessingEnabled()`

File: .gitmodules
Comment (mihnita) [reply .gitmodules:3]: I know. That part I did, and worked to test / fix things.

But I was unable to make the changes in `m2e-core-tests` show up in the `m2e-core` PR.
When I changed stuff and pushed it went in mihnita/m2e-core-tests.git, and I was able to make a PR from there.
But the PR from `m2e-core` ignored the changes in  `m2e-core-tests`

**Review by mihnita — APPROVED** [review]

File: .gitmodules
Comment (HannesWell) [reply .gitmodules]: > But the PR from `m2e-core` ignored the changes in `m2e-core-tests`

Changes of submodules are not shown in the container git, you only see that the ref-point to m2e-core-tests submodule has changed and you can only stage and commit that ref-pointer change. The staging view then looks like this:
grafik

But if you did that, maybe GH and Jenkins ignore changes in the .gitmodules of non-committers when checking out the submodules.

Comment (laeubi) [reply .gitmodules]: For me it has worked to just edit the `.gitmodules` file and never changed the ref pointer....

**Comment by HannesWell** [comment]
> > @mihnita can the MavenLaunchDelegegate be simplified further?
> 
> I can't imagine how. If we require a newer version of org.eclipse.ui.console, one that has ANSI, we can remove `isAnsiSupportInstalled()`, but nothing other than that.

OK, I just wanted to know if I missed something.

> 
> > IIRC you said that there is already a check per console if coloring is supported/enabled?
> 
> There is none. Right now the enable / disable status is global (for same for all consoles). And that is the one accessed with `preferencesService.getBoolean(...)` in `isAnsiProcessingEnabled()`

Then I understood that wrong and this is fine as it is.

**Pull request merged**
